servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DnsServiceDiscovererBuilder.java

```diff
@@ -30,6 +30,21 @@
  * resolve {@code A}, {@code AAAA}, {@code CNAME}, and {@code SRV} type queries.
  */
 public interface DnsServiceDiscovererBuilder {
+    /**
+     * Set the maximum size of the cache that is used to consolidate concurrent lookups for different hostnames.
```

Comment:
No, it's the max size of a hash map that holds `<Hostname, Future<Response>>` mappings. 
When a new request starts, it adds a new entry in the hash map. Then any other concurrent requests for the same hostname will take the future from the map.
When the Futute completes, it will remove the entry from the map.
Different hostnames will create new entries in the map.